### ISSUE TITLE
Confusing auth module in documentation

### ISSUE THREAD

**Issue by Nokitoo** [mage/mage#197]
Hi,

I find the auth module is confusing in the documentation.
It's not said that it's not enabled by default, but it's used in some examples, before explaining how to enable auth module.

It's the case in the examples of the [modules-methods](https://mage.github.io/mage/#module-methods) part, [using-async-await-node-7-6](https://mage.github.io/mage/#using-async-await-node-7-6) part, and the [login](https://mage.github.io/mage/#login) part.

**Comment by stelcheck** [comment]
Indeed. We should have fixed the documentation at the time we removed the auth built-in module from the list of default module.

I'll add a section on built-in modules and explain how to set up the auth module shortly.

**Comment by Nokitoo** [comment]
@stelcheck What do you think about this structure ?

- Modules
	- built-in modules
		- Explain what built-in modules are
		- Specify already activated modules on fresh mage install
		- Explain auth module is not activated by default and explain how to setup
	- File structure
	- Module setup and teardown


- Actors and session
	- Session module
	- Auth module  <= Remove setup explanation and add link to redirect on built-in modules
	- Logging in

**Comment by stelcheck** [comment]
Sounds good to me.
